conn.go

```diff
@@ -182,14 +196,29 @@ func (conn *Conn) Close() error {
 		conn.outLck.Unlock()
 		return nil
 	}
+
 	close(conn.out)
 	conn.closed = true
 	conn.outLck.Unlock()
+
+	close(conn.shutdownSigWorker)
+	conn.signalQueueLck.Lock()
+	// This is to avoid unnecessary wakeup of sigWorker
+	tmp := conn.signalQueueWakeUp
```

Comment:
Ok, that makes sense. And from my local testing, it appears safe to set a channel to nil when something is already `select`ing on it.